From 3ef0cd608434be58fed91b41f7a8ae68e00a41bb Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 23 May 2024 00:18:20 +0800
Subject: [PATCH 2/2] Expose this() for serialization

Fixes the following errors:

Error: constructor `inochi2d.core.nodes.drivers.Driver.this` is not
accessible from module `simplephysics`
Error: constructor `inochi2d.core.nodes.Node.this` is not accessible
from module `drivers`
---
 source/inochi2d/core/nodes/drivers/package.d | 6 +++++-
 source/inochi2d/core/nodes/package.d         | 7 ++++---
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/source/inochi2d/core/nodes/drivers/package.d b/source/inochi2d/core/nodes/drivers/package.d
index 9a46846..264e57a 100644
--- a/source/inochi2d/core/nodes/drivers/package.d
+++ b/source/inochi2d/core/nodes/drivers/package.d
@@ -17,7 +17,11 @@ public import inochi2d.core.nodes.drivers.simplephysics;
 */
 @TypeId("Driver")
 abstract class Driver : Node {
-private:
+public:
+
+    /**
+        Needed for deserialization
+    */
     this() { }
 
 protected:
diff --git a/source/inochi2d/core/nodes/package.d b/source/inochi2d/core/nodes/package.d
index 6090355..f7e2abd 100644
--- a/source/inochi2d/core/nodes/package.d
+++ b/source/inochi2d/core/nodes/package.d
@@ -74,8 +74,6 @@ void inClearUUIDs() {
 @TypeId("Node")
 class Node : ISerializable {
 private:
-    this() { }
-
     @Ignore
     Puppet puppet_;
 
@@ -216,8 +214,11 @@ package(inochi2d):
         this.puppet_ = puppet;
     }
 
+
 public:
 
+    this() { }
+
     /**
         Whether the node is enabled
     */
@@ -1051,4 +1052,4 @@ mixin template InNode(T) {
     static this() {
         inRegisterNodeType!(T);
     }
-}
\ No newline at end of file
+}
-- 
2.45.1

